src/main/scala/com/lucidworks/spark/ml/feature/LuceneTextAnalyzerTransformer.scala

```diff
@@ -122,10 +120,13 @@ class LuceneTextAnalyzerTransformer(override val uid: String) extends HasInputCo
     }
     StructType(schema.fields :+ new StructField($(outputCol), outputDataType, nullable = false))
   }
+
   override def transform(dataset: Dataset[_]): DataFrame = {
     val schema = dataset.schema
     val existingInputCols = $(inputCols).filter(schema.fieldNames.contains(_))
-    val outputSchema = transformSchema(schema)
+    if (analyzer == null || (analyzer.isEmpty && analyzerInitFailure.isEmpty)) {
+      validateAnalysisSchema($(analysisSchema)) // make sure analyzer has been instantiated
+    }
     val analysisFunc = udf { row: Row =>
       if (analyzer == null || (analyzer.isEmpty && analyzerInitFailure.isEmpty)) {
```

Comment:
Note that validation/instantiation only occurs if the analyzer has not yet been instantiated, so the cost per row should be minimal.

IIRC the issue solved by checking with every row is when Spark ships the `analysisFunc` udf to other nodes for parallel execution, the analyzer may not yet have been instantiated.  I don't know enough about the execution model to know if there's a one-time initialization step where this could be performed prior to row iteration - if there were such a thing, then the analyzer check could be moved out of the udf.